Sources/Support/StoreMessagesHelper.swift

```diff
@@ -57,16 +57,20 @@ actor StoreMessagesHelper: StoreMessagesHelperType {
     }
 
     func showStoreMessages(types: Set<StoreMessageType>) async {
+        var messagesToKeep: [StoreMessage] = []
```

Comment:
There may be a more elegant solution here, but modifying `self.deferredMessages` while iterating over it could lead to unexpected behavior, so I opted for this approach to avoid potential issues.

Reply:
Good call. I'd probably do it the opposite way: store the messages we processed, and delete them from the array at the end. That way we don't loose any messages that might've been added while we were processing.

We can also make the variable `Atomic<>` to make sure we handle threading correctly.

Reply:
Is that case possible since `StoreMessagesHelper` is an actor, and `deferMessagesIfNeeded()` is called in it's own separate Task (created [here in PurchasesOrchestrator](https://github.com/RevenueCat/purchases-ios/blob/b5e05d6ad8c5a5cc87aecb75af61e8b28afd6569/Sources/Purchasing/Purchases/PurchasesOrchestrator.swift#L169))?

Reply:
Good point, didn't realise it's an actor, we probably don't need the Atomic in that case.

However, there is an suspension point (await) inside the `deferredMessages`, and that could be an opportunity for the `storeMessagesProvider.messages` loop to process a new message, so I'd still store the messages we processed and delete them, rather than store the messages to keep.

Reply:
Makes sense - I'll make the change!